src/Microsoft.Management.Deployment/PackageManager.idl

```diff
@@ -1214,6 +1214,206 @@ namespace Microsoft.Management.Deployment
         String InstallationNotes { get; };
     }
 
+    /// IMPLEMENTATION NOTE: AddPackageCatalogOptions
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 11)]
+    runtimeclass AddPackageCatalogOptions
+    {
+        AddPackageCatalogOptions();
+
+        /// The name of the catalog to add.
+        String CatalogName;
+        /// The source URI of the catalog to add.
+        String SourceUri;
+        /// Type of the catalog to add.
+        String Type;
+        /// Additional arguments to pass to the catalog.
+        String Arguments;
+        /// The trust level of the catalog to add.
+        PackageCatalogTrustLevel TrustLevel;
+        /// Custom header to pass to the catalog.
+        String CustomHeader;
+    };
+
+    /// IMPLEMENTATION NOTE: AddPackageCatalogStatus
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 11)]
+    enum AddPackageCatalogStatus
+    {
+        Ok,
+        GroupPolicyError,
+        CatalogError,
+        InternalError,
+    };
+
+    /// IMPLEMENTATION NOTE: AddPackageCatalogResult
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 11)]
+    runtimeclass AddPackageCatalogResult
+    {
+        AddPackageCatalogStatus Status { get; };
+
+        /// Error codes
+        HRESULT ExtendedErrorCode { get; };
+    };
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 11)]
+    enum AddPackageCatalogProgressState

```

Comment:
This seems to be a copy-adapt of the package install state, but I don't think it properly represents the actual flow.  There is no queue unless we add one for COM.  The code is already multi-write resilient, so it shouldn't need a queue.

Also, we own the relevant actions, so there isn't a point at which we can't just revert state.

I don't think that this is really needed.  Discussing the larger progress type for all of these operations, I don't see a need for any of the *State enums.  We could probably just make the progress type `Double` for all of them.